src/NServiceBus.Core/Routing/FileBasedDynamicRouting/RoutingFileAccess.cs

```diff
@@ -0,0 +1,23 @@
+namespace NServiceBus
+{
+    using System.IO;
+    using System.Threading.Tasks;
+    using System.Xml;
+    using System.Xml.Linq;
+
+    class RoutingFileAccess : IRoutingFileAccess
+    {
+        public Task<XDocument> Load(string path)
+        {
+            XDocument doc;
+            using (var file = new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
```

Comment:
That does not look like `XDocument` ;-) I could not find any `async` overload of the `XDocument.Load` and I am not sure if asynchronously loading a file into a string and then calling `XDocument.Parse` will be better.